Dockerfile

```diff
@@ -0,0 +1,43 @@
+FROM python:3.8-slim-buster as builder
+
+ENV POETRY_VERSION=1.0.10
+
+# install poetry
+RUN pip install "poetry==$POETRY_VERSION"
+
+# set path of workdir in docker
+WORKDIR /app
+
+# copy my current dir to /app on WORKDIR
+COPY . /app
+
+# step to show my list of copied files and subdirectories
+RUN ls -la ./
+
+# turn off virtual envir creation from poetry as we're in docker
+# install poetry dependencies #COPY pyproject.toml poetry.lock ./
+RUN poetry config virtualenvs.create false \
+    && poetry install --no-interaction
+
+# Configure for production - multi build
+FROM python:3.8-slim-buster as production
+ENV POETRY_VERSION=1.0.10
+RUN pip install "poetry==$POETRY_VERSION"
+WORKDIR /app
+COPY --from=builder /app/ /app/ 
+RUN poetry config virtualenvs.create false \
+    && poetry install --no-interaction
+# set my port on container to 5001
+EXPOSE 5001
+CMD ["gunicorn", "--bind", "0.0.0.0:5001", "--log-level=debug", "app:create_app()"]
```

Comment:
For your production and development targets, you can actually build `FROM builder as production` rather than starting from the python:3.8 image again.

Then you only need to install poetry and run `poetry install` in the builder stage and not in production or development.

And skip `COPY --from=builder /app/ /app/ `

What you've done would be useful if you wanted a different base image for one build step that produces some files. But you're using the same base and repeating commands.

Reply:
Good point! I wasn't sure how to reuse builder. Thanks for that. Updated and both prod and dev are working fine .